pkg/apis/apps/v1beta2/defaults.go

```diff
@@ -71,15 +61,8 @@ func SetDefaults_StatefulSet(obj *appsv1beta2.StatefulSet) {
 		obj.Spec.UpdateStrategy.Type = appsv1beta2.OnDeleteStatefulSetStrategyType
 	}
 	labels := obj.Spec.Template.Labels
-	if labels != nil {
-		if obj.Spec.Selector == nil {
-			obj.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: labels,
-			}
-		}
-		if len(obj.Labels) == 0 {
-			obj.Labels = labels
-		}
+	if labels != nil && len(obj.Labels) == 0 {
+		obj.Labels = labels
```

Comment:
Delete this. You are setting the labels of the StatefulSet to the labels of the PodTemplateSpec. You do not want to do this.